### PR TITLE
Fixed first page link in pagination causes redirect to actual initial listing page.

### PR DESCRIPTION
**Pull request by Mauricio-Urrego**
### Ticket
- [Prevent pagination links to page 1 that contain the page suffix. ](https://app.asana.com/0/507013885678070/1200321868255267/f)

### Description
-

### CODE REVIEW

**Review by bogdanarizancu — APPROVED** [review]

File: src/Plugin.php [hunk 1]
```diff
@@ -203,6 +203,14 @@ public static function init() {
 
     // Disables output of related products if over-ride checkbox is enabled.
     add_action('woocommerce_after_single_product_summary', __NAMESPACE__ . '\WooCommerce::disableRelatedProducts', 0, 2);
+
+    /**
+     * Removes the `/page/1` suffix from archive URLs.
+     */
```
Comment (sun) [inline src/Plugin.php:209]: Only a inline comment (single line summary) here; PHPDoc block is only for the separate function definition.

File: src/Plugin.php [hunk 2]
```diff
@@ -203,6 +203,14 @@ public static function init() {
...
+    add_filter('paginate_links', function ($link) {
+      $link = preg_replace('@/page/1/?(\?|$)@', '$1', $link);
+      return user_trailingslashit($link);
+    });
```
Comment (sun) [inline src/Plugin.php:213]: Please move to a separate static function/callback.

**Review by sun — CHANGES_REQUESTED** [review]

File: src/Plugin.php [hunk 3]
```diff
@@ -203,6 +203,9 @@ public static function init() {
 
     // Disables output of related products if over-ride checkbox is enabled.
     add_action('woocommerce_after_single_product_summary', __NAMESPACE__ . '\WooCommerce::disableRelatedProducts', 0, 2);
+
+    // Removes the `/page/1` suffix from archive URLs.
```
Comment (sun) [inline src/Plugin.php:207]: Avoid backticks in PHP code whenever possible, they are the shorthand operator for executing shell commands:
https://www.php.net/manual/en/language.operators.execution.php

```suggestion
    // Removes the suffix '/page/1' from archive URLs.
```

File: src/Plugin.php [hunk 4]
```diff
@@ -346,4 +349,14 @@ public static function loadTextdomain() {
     load_plugin_textdomain(static::L10N, FALSE, static::L10N . '/languages/');
   }
 
+  /**
+   * Removes the `/page/1` suffix from archive URLs.
```
Comment (sun) [inline src/Plugin.php:353]:
```suggestion
   * Removes the suffix '/page/1' from archive URLs.
```

**Review by sun — APPROVED** [review]
@Mauricio-Urrego I corrected the PR title. It should state the original problem that is being addressed by the proposed change.

Please squash-merge using the corrected PR title as commit message.

**Pull request merged**